typescript/sdk/src/token/types.ts

```diff
@@ -31,28 +31,53 @@ export const NativeTokenConfigSchema = TokenMetadataSchema.partial().extend({
 export type NativeTokenConfig = z.infer<typeof NativeTokenConfigSchema>;
 export const isNativeTokenConfig = isCompliant(NativeTokenConfigSchema);
 
-export const CollateralTokenConfigSchema = TokenMetadataSchema.partial().extend(
-  {
+const sharedCollateralTokenConfig = {
+  // For xerc20lockbox and collateral vault, we use the token as the lockbox/vault respectively
+  token: z
+    .string()
+    .describe('Existing token address to extend with Warp Route functionality'),
+};
```

Comment:
should you have the types discriminated here?

Reply:
wdym?

Reply:
like this type is no different from the one below with the same `token` field